packages/boilerplate/src/router/index.js

```diff
@@ -0,0 +1,56 @@
+/* eslint-disable */
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import * as Rudy from '@respond-framework/rudy'
+
+const routesMap = {}
+
+class Router extends Component {
+  constructor(props) {
+    super(props);
+
+    const childs = Array.isArray(props.children) ? props.children : [props.children]
+
```

Comment:
You shall add invariant here, and throw readable error in case of no children provided.

Reply:
any ideas on how to evaluate the children ahead of the render? I cant resolve the dynamic route if i land on a page where addRoutes is called during the render - skipping the second render.

Reply:
In 99% cases, props would not be changed between constructor and render.
PS:
`const childs = React.Chuldren.toArray(props.children)` (https://reactjs.org/docs/react-api.html#reactchildrentoarray)

Reply:
thanksss!